### ISSUE TITLE
Automatic versioning breaks if mercurial repository is synced with a GIT upstream through hg-git extension

### ISSUE THREAD

**Issue by jaraco** [jaraco/hgtools#31]
Originally reported by: **Armin Gruner (BitBucket: [ArminGruner](http://bitbucket.org/ArminGruner), GitHub: [ArminGruner](http://github.com/ArminGruner))**

----------------------------------------

Hi,
if a local hg repo is pushed/pulled towards a GIT repository through the hg-git extension, bookmarks are used to synchronize which branches are pushed upstream:

```
#!shell
» hg tags
tip                               99:989ec00b2a11
default/master                    99:989ec00b2a11
```

Unfortunately, this breaks the check in hgtools/versioning.py:

```
#!python

if tags == ['tip'] and not self.is_modified():
        tags = self.get_parent_tags('tip')
```

hgtools always assumes the local sandbox has been modified. I've changed the code to just check whether the sandbox version is the tip-most version if the reported tags contain the "tip":

```
#!python

if 'tip' in tags and not self.is_modified():
        tags = self.get_parent_tags('tip')
```

Not sure whether this does not break any corner case, though..

Regards
Armin

----------------------------------------
- Bitbucket: https://bitbucket.org/jaraco/hgtools/issue/31

**Comment by jaraco** [comment]
Sounds reasonable to me. I'll roll out a patch.

**Comment by jaraco** [comment]
Released as 6.5.
